compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3544,9 +3544,38 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
         arrayTypeNode.elemtype = createTypeNode(arrayTypeDescriptorNode.memberTypeDesc());
         arrayTypeNode.dimensions = dimensionSize;
         arrayTypeNode.sizes = sizes;
+        for (BLangExpression member : sizes) {
```

Comment:
Didn't we discuss doing this at a later phase like code analysis?

IINM, at least initially, this class (previously BLangNodeTransformer) was meant to temporarily transform syntax tree nodes to AST nodes. If we do get rid of this, whatever validation we do here now will have to be redone in a separate phase.

Reply:
> Didn't we discuss doing this at a later phase like code analysis?

Yeah. But during another discussion, @gimantha suggested to add this fix into to semantic analyzer since it is related to type inferring.

Reply:
my concern was, not being able to use the parent node in semantic analyzer as it is not populated there. What would be much better solution? If BLangNodeTransformer is temporary we cannot keep adding stuff there too. But is the CodeAnalyser right place to move this change?

Reply:
Disallowing this isn't purely a type checking thing, right? We are saying that a particular kind of array type descriptor can be used only in certain contexts, which IMO can be part of code analysis.

At semantic analysis/type checker IMO we can continue with what we have, inferring the type where possible and preferably logging errors only when used in the allowed contexts (we may be able to use the new visitor for this and maybe even do it separately), and at code analysis analyze if it is used in a valid place?

Reply:
+1. Will update the PR. Thanks.

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34612/commits/dd4a64875eca77bcc52d1ceb139a8f36621b4cea